drivers/rtc/Kconfig.am1805

```diff
@@ -0,0 +1,28 @@
+#AMBIQ AM1805 RTC
+#
+#SPDX-License-Identifier: Apache-2.0
+#Copyright (c) 2023 Linumiz
+#Author: Sri Surya  <srisurya@linumiz.com>
+
+config RTC_AM1805
+	bool "AMBIQ AM1805 RTC driver"
+	depends on DT_HAS_AMBIQ_AM1805_ENABLED
+	select I2C if $(dt_compat_on_bus,$(DT_COMPAT_AMBIQ_AM1805),i2c)
```

Comment:
You should enable the driver by default and simply select I2C here, the 
`if $(dt_compat_on_bus,$(DT_COMPAT_AMBIQ_AM1805),i2c)` is only used
for devices that support multiple busses, the AM1805 is I2C only from what I can see :)

```
bool "AMBIQ AM1805 RTC driver"
	bool "AMBIQ AM1805 RTC driver"
	default y
	depends on DT_HAS_AMBIQ_AM1805_ENABLED
	select I2C
```